04_health/neonatal_health_code_2022.do

```diff
@@ -0,0 +1,701 @@
+}
+//loop replaces the value of new variable with the value from the unidentified counties within a state
+//repeated 255 times because that is the maximum number of counties in a state
+//end result is a single variable with unidentified counties value for every observation											
+
+replace lbw_births = unidentified_lbw if unidentified_county_flag == 1  //replaces all_birth with value from unidentified counties if an unidentified county, some remain missing due to suppressed data
+	drop unidentified_lbw
+
+* assign unidentified county value to missing counties: nomiss births
+sort subgroup state county
+by subgroup state: gen unidentified_nomiss = nomiss_births if county == 999		//create new variable with unidentified counties value
+
+forvalues i = 1/255 {
+	by subgroup state: replace unidentified_nomiss = unidentified_nomiss[_n+1] if missing(unidentified_nomiss)
+}
+//loop replaces the value of new variable with the value from the unidentified counties within a state
+//repeated 255 times because that is the maximum number of counties in a state
+//end result is a single variable with unidentified counties value for every observation											
+
+replace nomiss_births = unidentified_nomiss if unidentified_county_flag == 1  //replaces all_birth with value from unidentified counties if an unidentified county, some remain missing due to suppressed data
+
+*drop unidentified county observations because each unidentified county now has its own observation
+drop if county == 999
+drop county_name						// no longer need two county name variables
+	rename county_cross_name county_name
+	
+save "${health_data}neonatal_health_intermediate_raceth_`y2'.dta", replace
+
+//# *(6) create neonatal health share low birthweight metric
+
+//# lbw metric -  all births
+use "${health_data}neonatal_health_intermediate_all_`y2'.dta", clear
+*share lbw among nonmissing bw births 			// primary measure of lbw limiting the denominator to births with nonmissing birthweight data
+generate share_lbw_nomiss = lbw_births / nomiss_births
+	sum share_lbw_nomiss, detail
+	// assert share_lbw_nomiss < 1 // There are 9 zero nomiss_births values that result in missing share_lbw_nomiss values
+save "${health_data}neonatal_health_intermediate_all_`y2'.dta", replace
+	
+//# lbw metric -  race/ethnicity
+use "${health_data}neonatal_health_intermediate_raceth_`y2'.dta", clear
+*share lbw among nonmissing bw births 			// primary measure of lbw limiting the denominator to births with nonmissing birthweight data
+	generate share_lbw_nomiss = lbw_births/nomiss_births
+	replace share_lbw_nomiss = 0 if lbw_births == 0 & nomiss_births == 0	// TODO: Some question here - implication for the last assert statement in this block
+	
+	sum share_lbw_nomiss, detail
+	
+	assert share_lbw_nomiss < 1 if missing(suppressed_county_flag) & missing(unidentified_county_flag)
+	assert share_lbw_nomiss >= 0
+save "${health_data}neonatal_health_intermediate_raceth_`y2'.dta", replace
+
```

Comment:
Was the updated metric only supposed to be for all births and by race/ethnicity only? I see you imported mother's education, but I'm not seeing a share low birth weight metric by education